diffyscan/utils/explorer.py

```diff
@@ -15,36 +17,58 @@ def _get_contract_from_etherscan(token, etherscan_hostname, contract):
     if token is not None:
         etherscan_link = f"{etherscan_link}&apikey={token}"
 
-    response = fetch(etherscan_link)
+    response = fetch(etherscan_link).json()
 
     if response["message"] == "NOTOK":
-        logger.error("Failed", response["result"])
-        logger.error("Status", response.status_code)
-        logger.error("Response", response.text)
-        sys.exit(1)
+        raise ValueError(response["result"])
 
-    data = response["result"][0]
-    if "ContractName" not in data:
+    result = response["result"][0]
+    if "ContractName" not in result:
         _errorNoSourceCodeAndExit(contract)
 
-    contract_name = data["ContractName"]
-
-    json_escaped = data["SourceCode"].startswith("{{")
-    source_files = (
-        json.loads(data["SourceCode"][1:-1])["sources"].items()
-        if json_escaped
-        else json.loads(data["SourceCode"]).items()
-    )
-
-    return (contract_name, source_files)
+    solc_input = result["SourceCode"]
+
+    if solc_input.startswith("{{"):
+        return {
```

Comment:
Can we add a returning type to all such functions so it'll be easier to see that the return value is valid?